elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1217,8 +1217,10 @@ public Resource toResource(final Supplier<Map<String, Relationship>> relationshi
      * @param newValue the new value
      */
     protected void setValueChecked(String fieldName, Object newValue) {
-        checkFieldAwareDeferPermissions(UpdatePermission.class, fieldName, newValue, getValueUnchecked(fieldName));
+        final Object original = getValueUnchecked(fieldName);
+        checkFieldAwareDeferPermissions(UpdatePermission.class, fieldName, newValue, original);
```

Comment:
Let's have a version of checkPermission that takes a changeSpec - and use that here.